crates/subspace-service/src/transaction_pool.rs

```diff
@@ -0,0 +1,554 @@
+        self.inner.ready()
+    }
+
+    fn remove_invalid(&self, hashes: &[TxHash<Self>]) -> Vec<Arc<Self::InPoolTransaction>> {
+        self.inner.remove_invalid(hashes)
+    }
+
+    fn status(&self) -> PoolStatus {
+        self.inner.status()
+    }
+
+    fn futures(&self) -> Vec<Self::InPoolTransaction> {
+        self.inner.futures()
+    }
+
+    fn import_notification_stream(&self) -> ImportNotificationStream<TxHash<Self>> {
+        self.inner.import_notification_stream()
+    }
+
+    fn on_broadcasted(&self, propagations: HashMap<TxHash<Self>, Vec<String>>) {
+        self.inner.on_broadcasted(propagations)
+    }
+
+    fn hash_of(&self, xt: &TransactionFor<Self>) -> TxHash<Self> {
+        self.inner.hash_of(xt)
+    }
+
+    fn ready_transaction(&self, hash: &TxHash<Self>) -> Option<Arc<Self::InPoolTransaction>> {
+        self.inner.ready_transaction(hash)
+    }
+}
+
+#[async_trait]
+impl<Block, PoolApi> MaintainedTransactionPool for BasicPoolWrapper<Block, PoolApi>
+where
+    Block: BlockT,
+    PoolApi: ChainApi<Block = Block> + 'static,
+{
+    async fn maintain(&self, event: ChainEvent<Self::Block>) {
+        self.inner.maintain(event).await
+    }
+}
+
+pub fn new_full<Client, Block, DomainHeader>(
+    config: &Configuration,
+    task_manager: &TaskManager,
+    client: Arc<Client>,
+    sync_target_block_number: Arc<AtomicU32>,
+) -> sp_blockchain::Result<Arc<FullPool<Client, Block, DomainHeader>>>
+where
+    Block: BlockT,
+    <<<Block as BlockT>::Header as HeaderT>::Number as TryInto<u32>>::Error: std::fmt::Debug,
+    Client: ProvideRuntimeApi<Block>
+        + AuxStore
+        + BlockBackend<Block>
+        + HeaderBackend<Block>
+        + HeaderMetadata<Block, Error = sp_blockchain::Error>
+        + ExecutorProvider<Block>
+        + UsageProvider<Block>
+        + BlockIdTo<Block>
+        + Send
+        + Sync
+        + 'static,
+    DomainHeader: HeaderT,
+    Client::Api: TaggedTransactionQueue<Block>
+        + SubspaceApi<Block, FarmerPublicKey>
+        + FraudProofsApi<Block, DomainHeader>
+        + DomainsApi<Block, DomainHeader>,
+{
+    let prometheus = config.prometheus_registry();
+    let (fraud_proof_submit_sink, mut fraud_proof_submit_stream) = mpsc::unbounded_channel();
+    let pool_api = Arc::new(FullChainApiWrapper::new(
+        client.clone(),
+        prometheus,
+        task_manager,
+        sync_target_block_number,
+        fraud_proof_submit_sink,
+    )?);
+
+    let basic_pool = Arc::new(BasicPoolWrapper::with_revalidation_type(
+        config,
+        pool_api,
+        prometheus,
+        task_manager.spawn_essential_handle(),
+        client.clone(),
+    ));
+
+    let offchain_tx_pool_factory = OffchainTransactionPoolFactory::new(basic_pool.clone());
+
+    task_manager
+        .spawn_essential_handle()
+        .spawn_essential_blocking(
+            "consensus-fraud-proof-submitter",
+            None,
+            Box::pin(async move {
```

Comment:
Is it necessary to submit fraud proof with a dedicated worker? If so plz add some comments to explain.